src/NewProject/source_grid.cpp

```diff
@@ -148,20 +149,30 @@ void sourceGrid::AddFiles() {
 }
 
 void sourceGrid::AddDirectories() {
-  QString folder = QFileDialog::getExistingDirectory(
+  auto folder = QFileDialog::getExistingDirectory(
       this, tr("Select Directory"), "",
       QFileDialog::ShowDirsOnly | QFileDialog::DontResolveSymlinks);
-  QDir findFiles = folder;
-  QStringList files =
-      findFiles.entryList({"*.v", "*.sv", "*.vhd"}, QDir::Files);
-  foreach (QString fileName, files) {
-    QString suffix =
+
+  if (folder.isEmpty())  // The dialog was cancelled
+    return;
+  auto it = QDirIterator(folder, {"*.v", "*.sv", "*.vhd"}, QDir::NoFilter,
+                         QDirIterator::Subdirectories);
+  auto files =
+      std::vector<std::pair<QString, QString>>{};  // File names with directory
+                                                   // full paths
+  while (it.hasNext()) {
+    it.next();
```

Comment:
are you sure you don't skip first element here?

Reply:
I was also suspecting that, but the first element is kind'a root and is always empty.
Even Qt documentation example works this way:
while (it.hasNext()) {
     qDebug() << it.next();